libcaf_core/caf/detail/parser/read_uri.hpp

```diff
@@ -99,13 +99,13 @@ template <class State, class Consumer>
 void read_uri(State& ps, Consumer&& consumer) {
   // Local variables.
   std::string str;
+  std::string memory;
   uint16_t port = 0;
   // Replaces `str` with a default constructed string to make sure we're never
   // operating on a moved-from string object.
   auto take_str = [&] {
-    using std::swap;
     std::string res;
-    swap(str, res);
+    std::swap(str, res);
```

Comment:
Why did you change that? That's the canonical way to call `swap`. Since we are swapping strings, I'd also accept `str.swap(res)`.

Reply:
Basically, because I tought it was the same thing with one line less. I did my homework on the C++ reference manual, I will return it.